solution/dqt.domain/QualifiedTeachersService.cs

```diff
@@ -0,0 +1,30 @@
+﻿using dqt.datalayer.Model;
+using dqt.datalayer.Repository;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace dqt.domain
+{
+    public class QualifiedTeachersService : IQualifiedTeachersService
+    {
+        private readonly IRepository<QualifiedTeacher> _qualifiedTeachersRepository;
+
+        public QualifiedTeachersService(IRepository<QualifiedTeacher> repo)
+        {
+            _qualifiedTeachersRepository = repo;
+        }
+
+        public async Task<IEnumerable<QualifiedTeacher>> GetQualifiedTeacherRecords(string teacherReferenceNumber, string natonalInsuranceNumber)
+        {
+            var qts = await _qualifiedTeachersRepository.FindAsync(x => x.TeacherReferenceNumber == teacherReferenceNumber);
```

Comment:
Both TRN and NI DB calls can be joined by the OR operator.

Reply:
No, first we want to check with TRN then with NI only if no records found while checking with TRN